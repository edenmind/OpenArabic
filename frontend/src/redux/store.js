import {
  arabicSentenceReducer,
  arabicTextReducer,
  arabicWordsReducer,
  authorReducer,
  categoryReducer,
  englishSentenceReducer,
  englishTextReducer,
  englishWordsReducer,
  sourceReducer,
  textReducer,
  wordByWordReducer,
} from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    title: textReducer,
    category: categoryReducer,
    author: authorReducer,
    source: sourceReducer,
    englishText: englishTextReducer,
    arabicText: arabicTextReducer,
    arabicWords: arabicWordsReducer,
    englishWords: englishWordsReducer,
    arabicSentence: arabicSentenceReducer,
    englishSentence: englishSentenceReducer,
    wordByWord: wordByWordReducer,
  },
})
