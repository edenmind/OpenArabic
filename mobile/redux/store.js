import { configureStore, combineReducers } from '@reduxjs/toolkit'

import { categoriesReducer, textReducer, textsReducer, UIStateReducer, wordsReducer } from './reducers.js'

const rootReducer = combineReducers({
  categories: categoriesReducer,
  text: textReducer,
  texts: textsReducer,
  textLoading: UIStateReducer,
  textsLoading: UIStateReducer,
  arabicFontSize: UIStateReducer,
  englishFontSize: UIStateReducer,
  isTransliterationOn: UIStateReducer,
  arabicFontName: UIStateReducer,
  practicingWords: UIStateReducer,
  words: wordsReducer,
  isDarkMode: UIStateReducer
})

const store = configureStore({
  reducer: rootReducer
})

export { store }
