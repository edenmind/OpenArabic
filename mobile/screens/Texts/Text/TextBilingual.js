import "react-native-gesture-handler";
import React, { useState, useEffect } from "react";
import { ScrollView } from "react-native";
import { Title, Subheading, Paragraph } from "react-native-paper";
import * as api from "../../../services/ApiService";

export default function TextBilingual({ route }) {
  const { textId } = route.params;
  const [text, setText] = useState([{}]);

  useEffect(() => {
    async function fetchData() {
      console.log(textId);
      const textFromApi = await api.getText(textId);
      setText(textFromApi);
    }
    fetchData();

    console.log("This is the text we got!" + text);
  }, [textId]);

  return (
    <ScrollView>
      <Title>{text.title}</Title>
      <Subheading>{text.author}</Subheading>
      <Paragraph>{text.englishText}</Paragraph>
    </ScrollView>
  );
}
