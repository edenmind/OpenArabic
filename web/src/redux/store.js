import * as reducers from './reducers'

import { configureStore } from '@reduxjs/toolkit'

export default configureStore({
  reducer: {
    text: reducers.textReducer,
  },
})
