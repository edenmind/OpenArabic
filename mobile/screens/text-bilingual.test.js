import { describe, it, jest } from '@jest/globals'
import React from 'react'
import { Provider } from 'react-redux'
import { store } from '../redux/store.js'
import renderer from 'react-test-renderer'
import TextBilingual from './text-bilingual.js'
jest.useFakeTimers()

describe('<TextArabic />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <TextBilingual />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
