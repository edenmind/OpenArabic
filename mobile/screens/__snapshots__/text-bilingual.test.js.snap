// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArabic /> renders correctly 1`] = `
<RCTScrollView>
  <View>
    <Image
      source={
        {
          "uri": "abc",
        }
      }
      style={
        {
          "height": 250,
          "width": "100%",
        }
      }
    />
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "500",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 20,
                "letterSpacing": 0.15,
                "lineHeight": 30,
                "marginVertical": 2,
              },
              {
                "paddingTop": 55,
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
    >
      abc
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
                "letterSpacing": 0.5,
                "lineHeight": 24,
                "marginVertical": 2,
              },
              {
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
    />
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 12,
                "letterSpacing": 0.4,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              {
                "paddingBottom": 10,
                "textAlign": "center",
              },
            ],
          ],
        ]
      }
    />
    <View
      style={
        {
          "paddingBottom": 55,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "direction": "rtl",
              "flex": 1,
              "fontSize": 25,
              "lineHeight": 33,
              "paddingBottom": 10,
              "paddingLeft": 33,
              "paddingRight": 33,
              "paddingTop": 33,
              "writingDirection": "rtl",
            },
          ]
        }
      >
        a1
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "direction": "ltr",
              "flex": 1,
              "lineHeight": 25,
              "paddingLeft": 33,
              "paddingRight": 33,
              "writingDirection": "ltr",
            },
          ]
        }
      >
        e1
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": 64,
            "paddingLeft": 100,
            "paddingRight": 100,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#cccccc",
                    },
                  ]
                }
              >
                Vocabulary
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "direction": "rtl",
              "flex": 1,
              "fontSize": 25,
              "lineHeight": 33,
              "paddingBottom": 10,
              "paddingLeft": 33,
              "paddingRight": 33,
              "paddingTop": 33,
              "writingDirection": "rtl",
            },
          ]
        }
      >
        a2
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "direction": "ltr",
              "flex": 1,
              "lineHeight": 25,
              "paddingLeft": 33,
              "paddingRight": 33,
              "writingDirection": "ltr",
            },
          ]
        }
      >
        e2
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": 64,
            "paddingLeft": 100,
            "paddingRight": 100,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#cccccc",
                    },
                  ]
                }
              >
                Vocabulary
              </Text>
            </Text>
          </View>
        </View>
      </View>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "direction": "rtl",
              "flex": 1,
              "fontSize": 25,
              "lineHeight": 33,
              "paddingBottom": 10,
              "paddingLeft": 33,
              "paddingRight": 33,
              "paddingTop": 33,
              "writingDirection": "rtl",
            },
          ]
        }
      >
        a3
      </Text>
      <Text
        style={
          [
            {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            {
              "textAlign": "left",
            },
            {
              "direction": "ltr",
              "flex": 1,
              "lineHeight": 25,
              "paddingLeft": 33,
              "paddingRight": 33,
              "writingDirection": "ltr",
            },
          ]
        }
      >
        e3
      </Text>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": 64,
            "paddingLeft": 100,
            "paddingRight": 100,
            "shadowColor": "#000000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "disabled": false,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                [
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  {
                    "textAlign": "left",
                  },
                  [
                    {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    {
                      "textTransform": "uppercase",
                    },
                    {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <Text
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                    },
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "#cccccc",
                    },
                  ]
                }
              >
                Vocabulary
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
